apps/ex_wire/lib/ex_wire/kademlia/server.ex

```diff
@@ -65,6 +74,34 @@ defmodule ExWire.Kademlia.Server do
     {:reply, neighbours, state}
   end
 
+  def handle_info({:discovery_round, nodes}, %{routing_table: routing_table}) do
+    updated_table = Discovery.start(routing_table, nodes)
+
+    schedule_discovery_round(updated_table.discovery_round)
```

Comment:
I'm not sure if I'm not getting this right, but should this be passing a modified version of `nodes` as its second argument? 

This is how I see it: 
- In round `0`, `schedule_discovery_round` gets called from the `init` function so it gets the `nodes` and it passes them into the `Process.send_after` for round `1`. 
- In round `1`,  we hit this `handle_info({:discovery_round, nodes}` function and we update the table in line 78 and then schedule another round on his line (line 80). 
- In round `2`, when we hit this `handle_info` we'll get `{:discovery_round, []}`. 
- All subsequent rounds will have the same. 

Is that intentional? Do we only need the nodes for the first time we run `updated_table = Discovery.start(routing_table, nodes)`?

Reply:
@germsvel we always get discovery nodes from routing table. 

In the first round we do not have any nodes in table so we pass bootnodes and add them ourselves.  

in consecutive rounds we use nodes from routing table that came from previous discovery rounds that we did not use yet